### PR TITLE
Improve RTQ visibility and/or add ability to remove blocked object

### PR DESCRIPTION
**Pull request by jvoegele**
For background info see [JIRA: RIAK-1818] and #681 (RIAK-1818) (RIAK-1818).

New functions added to riak_repl2_rtq module:
- summarize/0 returns a list of tuples of the form `{Seq, Key, Size}`, one for each object currently present in the rtq.
- evict/1 removes an object from the rtq by `Seq` number, if present.
- evict/2 removes an object from the rtq by `Seq` number iff it matches the provided `Key`.

Currently these functions are not exposed through any command-line interface and must be executed from within the console.

### CODE REVIEW

File: src/riak_repl2_rtq.erl [hunk 1]
```diff
@@ -233,6 +236,26 @@ status() ->
...
 
+%% @doc Return summary data for the objects currently in the queue.
+%% The return value is a list of tuples of the form {SequenceNum, Key, Size}.
+-spec summarize() -> [{integer(), string(), integer()}].
+summarize() ->
+  gen_server:call(?SERVER, summarize, infinity).
+
+%% @doc If an object with the given Seq number is currently in the queue,
+%% evict it and return ok.
+evict(Seq) -> gen_server:call(?SERVER, {evict, Seq}, infinity).
```
Comment (macintux) [inline src/riak_repl2_rtq.erl]: Prefer to see `-spec` added to (at least) all exported functions.

File: test/riak_repl2_rtq_tests.erl [hunk 2]
```diff
@@ -2,9 +2,12 @@
 -compile(export_all).
 -include_lib("eunit/include/eunit.hrl").
 
+-define(setup_env, application:set_env(riak_repl, rtq_max_bytes, 10*1024*1024)).
```
Comment (cmeiklejohn) [inline test/riak_repl2_rtq_tests.erl]: Macros are usually capitalized for readability.

File: test/riak_repl2_rtq_tests.erl [hunk 3]
```diff
@@ -213,6 +274,38 @@ overload_test_() ->
...
+
+push_objects(Bucket, Keys) -> [push_object(Bucket, O) || O <- Keys].
+
+push_object(Bucket, Key) ->
+    RandomData = crypto:rand_bytes(1024 * 1024),
+    Obj = riak_object:new(Bucket, Key, RandomData),
+    riak_repl2_rtq:push(1, Obj),
+    Obj.
+
+push_random() ->
```
Comment (cmeiklejohn) [inline test/riak_repl2_rtq_tests.erl]: This function is not used.

File: test/riak_repl2_rtq_tests.erl [hunk 4]
```diff
@@ -213,6 +274,38 @@ overload_test_() ->
...
+    push_random(5).
+
+push_random(Num) ->
```
Comment (cmeiklejohn) [inline test/riak_repl2_rtq_tests.erl]: This function isn't used, and the code that inserts the queue results in an empty queue.  Why's it here?

```
(riak@127.0.0.1)9> riak_repl2_rtq:push(1, crypto:rand_bytes(1024 * 1024)).
ok
(riak@127.0.0.1)10> riak_repl2_rtq:summarize().     
```

File: test/riak_repl2_rtq_tests.erl [hunk 5]
```diff
@@ -213,6 +274,38 @@ overload_test_() ->
...
+    ?clean_env,
+    riak_repl_test_util:kill_and_wait(QPid).
+
+object_format() -> riak_core_capability:get({riak_kv, object_format}, v0).
+
+get_approximate_size(O) -> riak_object:approximate_size(object_format(), O).
+
+push_objects(Bucket, Keys) -> [push_object(Bucket, O) || O <- Keys].
+
+push_object(Bucket, Key) ->
```
Comment (cmeiklejohn) [inline test/riak_repl2_rtq_tests.erl:293]: Should this work?

```
(riak@127.0.0.1)16> riak_repl2_rtq:push(1,riak_object:new(<<"riak">>, <<"key">>, crypto:rand_bytes(1024 * 1024))).
ok
(riak@127.0.0.1)17> riak_repl2_rtq:summarize().                                                                   
[]
```

Comment (cmeiklejohn) [inline test/riak_repl2_rtq_tests.erl:293]: Oops, that's my fault.

I forgot to register a consumer.

```
(riak@127.0.0.1)22> Pid = whereis(riak_repl2_rtq).
<0.2240.0>
(riak@127.0.0.1)23> gen_server:call(Pid, {register, rtq_test}).
{ok,0}
(riak@127.0.0.1)24> riak_repl2_rtq:push(1,riak_object:new(<<"riak">>, <<"key">>, crypto:rand_bytes(1024 * 1024))).
ok
(riak@127.0.0.1)25> riak_repl2_rtq:summarize().                                                                   
[{1,<<"key">>,1048600}]
```

File: test/riak_repl2_rtq_tests.erl [hunk 6]
```diff
@@ -213,6 +274,38 @@ overload_test_() ->
 
     ]}.
 
+start_rtq() ->
+    ?setup_env,
+    %% application:start(lager),
```
Comment (cmeiklejohn) [inline test/riak_repl2_rtq_tests.erl]: Remove the commented code.

**Comment by cmeiklejohn** [comment]
Just some minor comments to address; :+1:, otherwise.

**Comment by cmeiklejohn** [comment]
Just to be clear, this PR should have been :+1:'d on the commit once the revisions were made and then merged by bors; bors will verify that the tests pass and the branch merges into an integration branch completely.

**Comment by jvoegele** [comment]
OK, got it. Thanks for clearing that up, @cmeiklejohn .

**Pull request merged**